vet-service/src/test/java/com/petclinic/vet/servicelayer/PhotoServiceImplTest.java

```diff
@@ -66,6 +68,30 @@ void getPhotoByValidVetId() {
                 })
                 .verifyComplete();
     }
+    @Test
+    void getDefaultPhotoByValidVetId() {
+        String photoName = "vet_default.jpg";
+        Photo savedDefaultPhoto = new Photo();
+        savedDefaultPhoto.setVetId(VET_ID);
+        savedDefaultPhoto.setFilename(photoName);
+        savedDefaultPhoto.setImgType("image/jpeg");
+        savedDefaultPhoto.setData(photoData);
+        when(photoRepository.save(any(Photo.class))).thenReturn(Mono.just(savedDefaultPhoto));
+
+        when(photoRepository.findByVetId(anyString())).thenReturn(Mono.just(photo));
+
+        Mono<PhotoResponseDTO> defaultPhotoMono = photoService.getDefaultPhotoByVetId(VET_ID);
+
+        StepVerifier
+                .create(defaultPhotoMono)
+                .consumeNextWith(image -> {
+                    assertNotNull(image);
+
+                    PhotoResponseDTO photo = defaultPhotoMono.block();
+                    assertEquals(photo.getVetId(), image.getVetId());
```

Comment:
In a later user story, this could be done in a more reactive way, but works fine.